lib/ffmprb/process.rb

```diff
@@ -58,6 +59,17 @@ def duck_audio(av_main_i, a_overlay_i, silence, av_main_o,
             roll in_main
 
             ducked_overlay_volume = {0.0 => volume_lo}
+
+            silence.each_cons(2) do |(silent, next_silent)|
+              silent_part = silent.overlaps || silent
```

Comment:
very sorry for delay; input should not be mutated, it should be pretty easy to reduce the input into a new array